src/Core/Environment.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.IO;
+
+namespace Bearded.Utilities
+{
+    /// <summary>
+    /// Enumerator for platforms.
+    /// </summary>
+    public enum Platform
+    {
+        /// <summary>
+        /// Windows platforms.
+        /// </summary>
+        Windows,
+        /// <summary>
+        /// Linux platforms.
+        /// </summary>
+        Linux,
+        /// <summary>
+        /// Mac platforms (OSX).
+        /// </summary>
+        OSX
+    }
+
+    /// <summary>
+    /// Collection of platform-specific functions.
+    /// </summary>
+    public static class Environment
+    {
+        /// <summary>
+        /// The platform the application is currently running on.
+        /// </summary>
+        public static Platform CurrentPlatform
```

Comment:
This might be nicer with caching. At least I can see no way in which the Platform can ever change at runtime.